client/src/components/CreateMealForm.js

```diff
@@ -0,0 +1,248 @@
+    reValidateMode: "onChange",
+    validateCriteriaMode: "all",
+  });
+
+  useEffect(() => {
+    register("image");
+    setValue("image", image);
+  });
+
+  const [image, setImage] = useState();
+  const [previewImage, setPreviewImage] = useState("");
+
+  const fields = [
+    {
+      name: "name",
+      label: "Meal Name",
+      defaultValue: meal.name,
+      validation: {
+        required: "Meal Name is required.",
+      },
+    },
+    {
+      name: "price",
+      label: "Price",
+      defaultValue: meal.price,
+      validation: {
+        required: "Price is required.",
+        pattern: {
+          value: /\d+/,
+          message: "Please supply a proper price",
+        },
+      },
+    },
+    {
+      name: "servings",
+      label: "Servings",
+      defaultValue: meal.servings,
+      validation: {
+        required: "Servings is required.",
+        pattern: {
+          value: /^\d+$/,
+          message: "Please supply a proper serving size",
+        },
+      },
+    },
+    {
+      name: "ingredients",
+      label: "Ingredients",
+      defaultValue: meal.ingredients,
+    },
+    {
+      name: "required_items",
+      label: "Required Items",
+      defaultValue: meal.required_items,
+    },
+  ];
+
+  return (
+    <>
+      <form
+        onSubmit={handleSubmit(onFormSubmit)}
+        className={classes.form}
+        noValidate
+      >
+        {!user.isChef && (
+          <>
+            <Typography variant="h5" component="h3">
+              Select a Cuisine for your Chef Profile
+            </Typography>
+            <Box display="inline" wrap="flex-wrap">
+              {cuisines.map((c, i) => (
+                <Button
+                  color={c.name === selectedCuisine ? "primary" : "default"}
+                  variant="contained"
+                  className={classes.cuisineButton}
+                  onClick={() => setSelectedCuisine(c.name)}
+                  key={i}
+                >
+                  {c.name}
+                </Button>
+              ))}
+            </Box>
+          </>
+        )}
+        <Typography variant="h5" component="h3" className={classes.mealTitle}>
+          {!user.isChef
+            ? "Add your first meal"
+            : mealId
+            ? "Edit your meal"
+            : "Add a meal"}
+        </Typography>
+        <img
+          className={classes.image}
+          src={previewImage || meal.image || imagePlaceholder}
+          alt="meal"
+        />
+
+        <Dropzone setImageFile={setImage} setPreviewImage={setPreviewImage} />
+
+        {fields.map(
+          (
+            {
+              name,
+              label,
+              defaultValue = "",
+              validation,
+              multiline = false,
+              rows,
+            },
+            index
+          ) => (
+            <FormControl key={index}>
+              <TextField
+                variant="outlined"
+                margin="normal"
+                required
```

Comment:
I noticed when you demo, that all the fields are shown as required, when they are not required. We should make that match (and if you remove noValidate from the form, we can actual just use HTML forms to validate)